specification/dartLangSpec.tex

```diff
@@ -1868,7 +1853,25 @@ \subsection{Formal Parameters}
 The formal parameter scope of a generic function $f$ is enclosed in
 the type parameter scope of $f$.
 Every formal parameter introduces a local variable into
-the formal parameter scope.
+the formal parameter scope,
+with the declared type which is the declared type of the formal parameter.
+If no type annotation is given for the formal parameter,
+the declared type is considered to be the type \DYNAMIC.
+
+\commentary{%
+Type inference is not yet specified in this document,
+and is assumed to have taken place already
+(\ref{overview}).
+In particular, some or all formal parameters of an instance method declaration
+may have received a declared type based on override inference,
+in which case it is not considered to be omitted.
+%% TODO(eernst): Change to \ref{} when inference spec is added.
+Briefly, override inference yields a type computed from the declared types of
+the corresponding parameter of member signatures in superinterfaces,
+or it causes a compile-time error if said types can not be reconciled.%
```

Comment:
Do you also want to mention inference from initializing formals here too since you include constructors under this umbrella now?

Reply:
Yes, thanks!